modin/experimental/cloud/meta_magic.py

```diff
@@ -0,0 +1,126 @@
+
+    def __getattribute__(self, name):
+        if name in _LOCAL_ATTRS:
+            # never proxy special attributes, always get them from the class type
+            res = object.__getattribute__(self, name)
+        else:
+            try:
+                # Go for proxying class-level attributes first;
+                # make sure to check for attribute in self.__dict__ to get the class-level
+                # attribute from the class itself, not from some of its parent classes.
+                # Also note we use object.__getattribute__() to skip any potential class-level __getattr__
+                res = object.__getattribute__(self, "__dict__")[name]
+            except KeyError:
+                try:
+                    res = object.__getattribute__(self, name)
+                except AttributeError:
+                    frame = sys._getframe()
+                    try:
+                        is_inspect = frame.f_back.f_code.co_filename == inspect.__file__
+                    except AttributeError:
+                        is_inspect = False
+                    finally:
+                        del frame
+                    if is_inspect:
+                        # be always-local for inspect.* functions
+                        res = super().__getattribute__(name)
+                    else:
+                        try:
+                            remote = object.__getattribute__(
+                                object.__getattribute__(self, "__real_cls__"),
+                                "__wrapper_remote__",
+                            )
+                        except AttributeError:
+                            # running in local mode, fall back
+                            res = super().__getattribute__(name)
+                        else:
+                            res = getattr(remote, name)
+        try:
+            # note that any attribute might be in fact a data descriptor,
+            # account for that
+            getter = res.__get__
+        except AttributeError:
+            return res
+        return getter(None, self)
+
+
+_KNOWN_DUALS = {}
+
+
+def make_wrapped_class(local_cls, rpyc_wrapper_name):
+    namespace = {
+        "__real_cls__": None,
+        "__new__": None,
+        "__module__": local_cls.__module__,
+    }
+    result = RemoteMeta(local_cls.__name__, (local_cls,), namespace)
+
+    def make_new(__class__):
+        """
+        Define a __new__() with a __class__ that is closure-bound, needed for super() to work
+        """
+
+        def __new__(cls, *a, **kw):
+            if cls is result and cls.__real_cls__ is not result:
+                return cls.__real_cls__(*a, **kw)
+            return super().__new__(cls)
+
+        __class__.__new__ = __new__
+
+    make_new(result)
+    setattr(sys.modules[local_cls.__module__], local_cls.__name__, result)
+    _KNOWN_DUALS[local_cls] = result
+
+    def update_class(_):
```

Comment:
I don't understand why `update_class` needs `_` parameter.

Reply:
That's how current PubSub works - this callback receives the instance of `Publisher` which called it. We don't use it here, so I named the parameter `_`.